src/actions/plugins/surround.ts

```diff
@@ -120,7 +111,16 @@ class YankSurroundOperator extends SurroundOperator {
         rightEdge,
         cursorIndex: multicursorIndex,
       };
-    }
+    };
+    // then collect ranges for all cursors
+    const multicursorIndex = this.multicursorIndex ?? 0;
+    vimState.surround!.edges.push(getYankRanges());
+    vimState.cursorStartPosition = start;
+    // when called from visual operator, use end for stop to keep visual selection
+    vimState.cursorStopPosition = vimState.currentMode === Mode.Visual ? end : start;
+    await vimState.setCurrentMode(Mode.SurroundInputMode);
+
+    return;
```

Comment:
This return is now (more obviously) redundant

Reply:
fixed.